apps/server/src/utils/fileManagement.ts

```diff
@@ -114,3 +109,20 @@ export async function dockerSafeRename(oldPath: PathLike, newPath: PathLike) {
   await copyFile(oldPath, newPath);
   await unlink(oldPath);
 }
+
+/**
+ * finds potential file index number in our (*) format and increments
+ */
+export function incrementProjectNumber(path: string): string {
```

Comment:
suggestion: could we simplify this logic with a more generic algorithm?

- look for last opening parenthesis `(`
- if this exists, look for closing last parenthesis `)`

- try to parse the content inside the parenthesis as a number
- add `+1` to the number
- replace the new content where the old one was

- if any of the above fails or the order of parenthesis is invalid (opening is after closing) is invalid add the (1) before the last `.`

Reply:
if the `( )` is not at the end I dont think we can consider it ours?